src/error.rs

```diff
@@ -262,6 +269,85 @@ fn from_sslerror_ref(err: &ssl::Error) -> Error {
     }
 }
 
+#[cfg(feature = "snappy")]
+fn from_snap_error_ref(err: &::snap::Error) -> Error {
+    match err {
+        &::snap::Error::TooBig { given, max } => {
```

Comment:
If you don't care about all the individual errors, then just turn it into a `Box<Error>` or a `String`.